nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/ui/web/auth/NuxeoAuthenticationFilter.java

```diff
@@ -777,10 +779,17 @@ public boolean saveRequestedURLBeforeRedirect(HttpServletRequest httpRequest, Ht
     }
 
     /**
-     * The requested URL is like the requested page but also includes the query string, except without conversation id.
+     * The requested URL is like the requested page BUT is not decoded AND also includes the query string (except
+     * without conversation id).
      */
     public static String getRequestedUrl(HttpServletRequest request) {
         String path = getRequestedPage(request);
+        // re-encode path, as it's what the caller expects
+        try {
+            path = new URI(null, null, path, null).toASCIIString();
```

Comment:
I suppose there is no simpler built-in way to encode the path, and `URLEncoder#encode` is not what we want her?

Reply:
Yes this is the best I found. `URLEncoder` is for encoding the query part of URLs (or the body of form POSTs).